ArcGISRuntimeSDKQt_CppSamples/Maps/MobileMap_SearchAndRoute/MobileMap_SearchAndRoute.cpp

```diff
@@ -0,0 +1,366 @@
+// [WriteFile Name=MobileMap_SearchAndRoute, Category=Maps]
+// [Legal]
+// Copyright 2016 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "MobileMap_SearchAndRoute.h"
+
+#include "Map.h"
+#include "Stop.h"
+#include "Graphic.h"
+#include "RouteTask.h"
+#include "TextSymbol.h"
+#include "LocatorTask.h"
+#include "RouteResult.h"
+#include "MapQuickView.h"
+#include "GeocodeResult.h"
+#include "SimpleRenderer.h"
+#include "GraphicsOverlay.h"
+#include "RouteParameters.h"
+#include "SimpleLineSymbol.h"
+#include "MobileMapPackage.h"
+#include "PictureMarkerSymbol.h"
+#include "ReverseGeocodeParameters.h"
+
+#include <QDir>
+#include <QQmlProperty>
+#include <QFileInfoList>
+#include <QFile>
+
+using namespace Esri::ArcGISRuntime;
+
+MobileMap_SearchAndRoute::MobileMap_SearchAndRoute(QQuickItem* parent):
+    QQuickItem(parent),
+    m_map(nullptr),
+    m_mapView(nullptr),
+    m_calloutData(nullptr),
+    m_bluePinSymbol(nullptr),
+    m_currentRouteTask(nullptr),
+    m_currentLocatorTask(nullptr),
+    m_stopsGraphicsOverlay(nullptr),
+    m_routeGraphicsOverlay(nullptr),
+    m_canRoute(false),
+    m_canClear(false),
+    m_isGeocodeInProgress(false)
+{
+}
+
+MobileMap_SearchAndRoute::~MobileMap_SearchAndRoute()
+{
+}
+
+void MobileMap_SearchAndRoute::componentComplete()
+{
+    QQuickItem::componentComplete();
+
+    // find QML MapView component
+    m_mapView = findChild<MapQuickView*>("mapView");
+    m_mapView->setWrapAroundMode(WrapAroundMode::Disabled);
+
+    m_fileInfoList = QDir(QDir::homePath() + "/ArcGIS/Runtime/Data/mmpk/").entryInfoList();
```

Comment:
This might cause some problems on ios.  It is better to use paths from qml and expose them as properties and then using them from c++.  Some of the samples have that mechanism.  You can look at those for examples.